sdk/typespec/typespec_client_core/src/http/headers/mod.rs

```diff
@@ -69,6 +93,28 @@ impl Headers {
         Self::default()
     }
 
+    /// Gets the headers represented by `H`, or return an error if the header is not found.
+    pub fn get<H: FromHeaders>(&self) -> crate::Result<H> {
+        match H::from_headers(self) {
+            Ok(Some(x)) => Ok(x),
+            Ok(None) => Err(crate::Error::with_message(
+                ErrorKind::DataConversion,
+                || format!("required headers not found: {}", H::HEADER_NAMES.join(", ")),
```

Comment:
Nit: most impls will likely just have one header, so maybe:
```suggestion
                || format!("required header(s) not found: {}", H::HEADER_NAMES.join(", ")),
```

Reply:
I'll just do the `if` to do plural right. I'm not a fan of using `(s)` for this kind of thing.

Reply:
But, on the other hand, it makes a consistent message to search for in logs, traces, stack traces, etc. In normal UI messages I don't like cop-outs like that either, but this is basically a debug/dev error. Consistency is king.

E.g., some parts of Windows Installer error messages are localized while others aren't. Those in the know search for "value 3" because it's the only part of a typical error message not localized when looking for fatal errors. You don't want that for our customers, do you?

Reply:
Good point, I'll revert to your suggestion.